overlord/devicestate/devicemgr.go

```diff
@@ -780,29 +792,53 @@ func (m *DeviceManager) Systems() ([]System, error) {
 		return nil, ErrNoSystems
 	}
 
-	var systems []System
+	var systems []*System
 	for _, fpLabel := range systemLabels {
 		label := filepath.Base(fpLabel)
-		model, brand, err := systemSeedModelAndBrand(label)
+		system, err := systemFromSeed(label)
 		if err != nil {
 			// TODO:UC20 add a Broken field to the seed system like
 			// we do for snap.Info
 			logger.Noticef("cannot load system %q seed: %v", label, err)
 			continue
 		}
-		systems = append(systems, System{
-			// TODO:UC20 check if current installation was done with that
-			// system
-			Current: false,
-			Label:   label,
-			Model:   model,
-			Brand:   brand,
-			// TODO:UC20: fill actions
-		})
+		// TODO:UC20 check if current installation was done with that
+		// system
+		system.Current = false
+		systems = append(systems, system)
 	}
 	return systems, nil
 }
 
+// RequestSystemAction request provided system to be run in a given mode. A
+// system reboot will be requested when the request can be successfully carried
+// out.
+func (m *DeviceManager) RequestSystemAction(systemLabel string, action SystemAction) error {
+	systemSeedDir := filepath.Join(dirs.SnapSeedDir, "systems", systemLabel)
+	if _, err := os.Stat(systemSeedDir); err != nil {
+		return err
+	}
+	system, err := systemFromSeed(systemLabel)
+	if err != nil {
+		return fmt.Errorf("cannot load seed system: %v", err)
+	}
+
+	var sysAction *SystemAction
+	for _, act := range system.Actions {
+		if action.Mode == act.Mode {
+			sysAction = &act
+			break
+		}
+	}
+	if sysAction == nil {
+		return fmt.Errorf("cannot request unsupported mode %q for system %q",
+			action.Mode, systemLabel)
+	}
+
+	// TODO:UC20 request action
```

Comment:
This comment could be expanded a little bit, I suppose this means that we actually set the right boot option and schedule a reboot?

Reply:
yeah, I'll update the comment